src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -573,9 +573,19 @@ public TResult[] ToArray()
                 }
                 return results;
             }
+
+            public List<TResult> ToList()
+            {
+                var results = new List<TResult>(_source.Length);
```

Comment:
There's something interesting with List to consider here, and in all of the other cases below where we presize the list.  Previously, we'd just use the normal ```List<T>``` ctor, which would check for ```ICollection<T>``` and use its count if it could or else just use the normal list growth algorithm, starting at 0, growing to 4 for the first element, and then doubling after that.   Since all of these cases didn't implement ```ICollection<T>```, we would get the standard 0/4/doubling behavior.  And that means that, for example, if you did ```new int[2] { ... }.Select(...).ToList()```, you'd get back a list with a capacity of 4 rather than 2, which means that if the caller added an additional item or two into the list, it wouldn't result in the list needing to grow.

It's probably relatively rare to add items to the returned list, so what you have here is probably the right thing to do, but I wanted to mention it.